packages/framework/src/Model/Order/OrderFacade.php

```diff
@@ -661,20 +661,32 @@ protected function refreshOrderItemsWithoutTransportAndPayment(Order $order, Ord
         foreach ($orderData->getNewItemsWithoutTransportAndPayment() as $newOrderItemData) {
             $this->calculateOrderItemDataPrices($newOrderItemData);
 
-            $newOrderItem = $this->orderItemFactory->createProduct(
-                $order,
-                $newOrderItemData->name,
-                new Price(
-                    $newOrderItemData->priceWithoutVat,
-                    $newOrderItemData->priceWithVat
-                ),
-                $newOrderItemData->vatPercent,
-                $newOrderItemData->quantity,
-                $newOrderItemData->unitName,
-                $newOrderItemData->catnum
-            );
-            if (!$newOrderItemData->usePriceCalculation) {
-                $newOrderItem->setTotalPrice(new Price($newOrderItemData->totalPriceWithoutVat, $newOrderItemData->totalPriceWithVat));
+            if (method_exists($this->orderItemFactory, 'createProductFromOrderItemData')) {
+                $this->orderItemFactory->createProductFromOrderItemData($order, $newOrderItemData);
+            } else {
+                $message = sprintf(
+                    'Adding new OrderItem via "%s" is deprecated since SSFW 8.1, implement "%s" instead.'
+                    . 'Interface "%s" will be changed in SSFW 9.',
+                    get_class($this->orderItemFactory) . '::createProduct()',
```

Comment:
get_class returns FQCN, I would use only 
```suggestion
                    'OrderItemFactory::createProduct()',
```